src/j1cli.js

```diff
@@ -0,0 +1,191 @@
+    .option('--entity', 'Specifies entity operations.')
+    .option('--relationship', 'Specifies relationship operations.')
+    .option('--alert', 'Specifies alert rule operations.')
+    .option('-f, --file <dir>', 'Input JSON file.')
+    .parse(process.argv);
+
+  try {
+    const data = await validateInputs();
+    const j1Client = await initializeJ1Client();
+
+    if (program.query) {
+      const res = await j1Client.queryV1(program.query);
+      console.log(JSON.stringify(res, null, 2));
+    }
+    else {
+      switch (program.operation) {
+        case 'create':
+          if (program.entity) {
+            await createEntities(j1Client, Array.isArray(data) ? data : [data]);
+          } else if (program.alert) {
+            await createAlerts(j1Client, Array.isArray(data) ? data : [data]);
+          }
+          break;
+        case 'update':
+          if (program.entity) {
+            await updateEntities(j1Client, Array.isArray(data) ? data : [data]);
+          } else if (program.alert) {
+            await updateAlerts(j1Client, Array.isArray(data) ? data : [data]);
+          }
+          break;
+        default:
+          console.log(`Unsupported operation: ${program.operation}`);
+      }
+    }
+  } catch (err) {
+    error.fatal(`Unexpected error: ${err}`);
+  }
+  console.log('Done!');
+}
+
+// ensure user supplied necessary params
+async function validateInputs () {
+  console.log('Validating inputs...');
+  if (!program.account || program.account === '') {
+    error.fatal('Missing -a|--account flag!', EUSAGEERROR);
+  }
+
+  let data;
+
+  if (!program.query || program.query === '') {
+    if (!program.entity && !program.relationship && !program.alert) {
+      error.fatal('Must specify a query (using -q|--query) or an entity/relationship/rule operation (--entity or --relationship or --alert)', EUSAGEERROR);
+    } else if (!program.operation || program.operation === '') {
+      error.fatal('Must specify command action for entity or rule operation (-c|--command <create|update|delete>)', EUSAGEERROR);
+    } else if (!program.file || program.file === '') {
+      error.fatal('Must specify input JSON file with -f|--file)', EUSAGEERROR);
+    } else if (!fs.existsSync(program.file)) {
+      error.fatal(`Could not find input JSON file (${program.file}). Specify the correct path with '-f|--file'.`);
+    } else {
+      data = jParse(program.file);
+      if (!data) {
+        error.fatal(`Could not parse input JSON file (${program.file}).`);
+      }
+    }
+  }
+  
+  if ((!program.key || program.key === '') && !J1_API_TOKEN) {
+    if (!program.user || program.user === '') {
+      error.fatal('Must authenticate with either the API key (using -k|--key) or username/password (using -u|--user)', EUSAGEERROR);
+    } else {
+      await gatherPassword();
+    }
+  }
+
+  return data;
+}
+
+function jParse (file) {
+  let data;
+  try { data = JSON.parse(fs.readFileSync(file, 'utf8')); } catch (err) { return null; }
```

Comment:
Similar to the above, you can `promisify` the `fs.readFile` function and `await` on it.